crypto-core/src/main/java/com/palantir/crypto/io/DecryptingSeekableInput.java

```diff
@@ -115,7 +146,7 @@ public long getPos() throws IOException {
     @Override
     public int read(byte[] buffer, int offset, int length) throws IOException {
         int bytesRead = decryptedStream.read(buffer, offset, length);
-        if (bytesRead != -1) {
+        if (bytesRead != 0) {
```

Comment:
This seems off, -1 indicates end of stream so The previous should be proper behavior.